pkg/query/storeset.go

```diff
@@ -341,16 +345,23 @@ func (s *StoreSet) updateStoreStatus(store *storeRef, err error) {
 	s.storesStatusesMtx.Lock()
 	defer s.storesStatusesMtx.Unlock()
 
-	now := time.Now()
-	s.storeStatuses[store.addr] = &StoreStatus{
-		Name:      store.addr,
-		LastError: err,
-		LastCheck: now,
-		Labels:    store.labels,
-		StoreType: store.storeType,
-		MinTime:   store.minTime,
-		MaxTime:   store.maxTime,
+	status := StoreStatus{Name: store.addr}
+	prev, ok := s.storeStatuses[store.addr]
+	if ok {
+		status = *prev
+	}
+
+	status.LastError = err
+	status.LastCheck = time.Now()
+
+	if err == nil {
```

Comment:
This is quite weird to me. Essentially not consistent. What if
1. We have success check and then fail. Why we have obsolete label, type ,min, max?
2. We have err as first think. We have it empty.

That's kind of inconsistent?

Reply:
Might not be the more explicit way of doing things, didn't want to add too many cases. Here's my logic:

* Query with a `--store=foo.local` argument
* Healthy first, status populated
* It's now broken, after a few it will be dropped
* Next updates won't have component, labels and time info so would overwrite existing status info

Does that make sense ?

For the second point, not sure I follow you

Reply:
I was talking about consistent error handling. You have two basic cases.

1 as talked through
2 which is `store=foo.local` was never healthy. 

I think I am missing the fact that 2 does not exists because it was never healthy ;p Ignore me then.